lib/Tools/LayoutUtils.cpp

```diff
@@ -185,6 +185,81 @@ LinearLayout identityStandardND(StringAttr inDimName, ArrayRef<unsigned> shape,
   return ret;
 }
 
+LinearLayout zerosLike(const LinearLayout &layout) {
+  auto bases = layout.getBases();
+  for (auto &basis : bases) {
+    for (auto &vec : basis.second) {
+      for (auto &val : vec) {
+        val = 0;
+      }
+    }
+  }
+
+  SmallVector<std::pair<StringAttr, int32_t>> outDims;
+  for (auto outDim : layout.getOutDimNames()) {
+    outDims.emplace_back(outDim, layout.getOutDimSize(outDim));
+  }
+  return LinearLayout(std::move(bases), std::move(outDims),
+                      /*requireSurjective=*/false);
+}
+
+std::optional<ColumnAction> actionDivideLeft(const LinearLayout &A,
+                                             const LinearLayout &B) {
```

Comment:
IIUC this is more like `findRegPermutationThatDividesLeft`?

Reply:
sure, I was just looking for a shorter name as this function is going to be used quite a bit. I was also thinking of packing this + the division and action on the regs into a helper function. Might do that tomorrow.

Reply:
chose a shorter but representative name